Source/WebKit/Shared/Cocoa/CoreIPCNSURLRequest.h

```diff
@@ -0,0 +1,166 @@
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY APPLE INC. AND ITS CONTRIBUTORS ``AS IS''
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+ * THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL APPLE INC. OR ITS CONTRIBUTORS
+ * BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
+ * THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#pragma once
+
+#if PLATFORM(COCOA)
+
+#import "CoreIPCBoolean.h"
+#import "CoreIPCData.h"
+#import "CoreIPCDate.h"
+#import "CoreIPCNumber.h"
+#import "CoreIPCPlistArray.h"
+#import "CoreIPCPlistDictionary.h"
+#import "CoreIPCString.h"
+#import "CoreIPCURL.h"
+
+#import <wtf/RetainPtr.h>
+#import <wtf/Vector.h>
+
+OBJC_CLASS NSURLRequest;
+
+namespace WebKit {
+
+enum class NSURLRequestCachePolicy : uint8_t {
+    UseProtocolCachePolicy,
+    ReloadIgnoringLocalCacheData,
+    ReturnCacheDataElseLoad,
+    ReturnCacheDataDontLoad,
+    ReloadIgnoringLocalAndRemoteCacheData,
+    ReloadRevalidatingCacheData,
+};
+
+enum class NSURLRequestNetworkServiceType : uint8_t {
+    Default,
+    OpaqueValue0,
+    OpaqueValue1,
+    Background,
+    OpaqueValue2,
+    OpaqueValue3,
+    OpaqueValue4,
+    OpaqueValue5,
+    AVStreaming,
+    OpaqueValue6,
+    OpaqueValue7,
+    OpaqueValue8
+};
+
+enum class NSURLRequestAllowedProtocolTypes : uint8_t {
+    OpaqueValue0,
+    OpaqueValue1 = (1 << 0),
+    OpaqueValue2 = (1 << 1),
+    OpaqueValue3 = (1 << 2),
+    OpaqueValue4 = (1 << 3),
+    OpaqueValue5 = (1 << 4),
+    OpaqueValue6 = (1 << 5),
+    OpaqueValue7 = (1 << 6),
+    OpaqueValue8 = (1 << 7),
+};
+
+enum class NSURLRequestAttribution : uint8_t {
+    Developer,
+    User
+};
+
+enum class NSURLRequestFlags : int16_t {
+    ShouldHandleCookies = (1 << 1),
+    NetworkServiceType = (1 << 2),
+    AllowsCellular = (1 << 3),
+    PreventsIdleSleep = (1 << 4),
+    ShouldPipeline = (1 << 6),
+    CachePolicy = (1 << 7),
+    Timeout = (1 << 8),
+    ProxyDict = (1 << 9),
+    SSLProperties = (1 << 10),
+    ShouldStartSynchronously = (1 << 12)
+};
+
+struct CoreIPCNSURLRequestData {
+
+    using BodyParts = std::variant<CoreIPCString, CoreIPCData>;
+    using Header = std::pair<String, std::variant<String, Vector<String>>>;
+
+    std::optional<CoreIPCPlistDictionary> protocolProperties;
+    bool isMutable;
```

Comment:
Let's give all the bools, doubles, enums and ints default values of 0/false to prevent the possibility of using uninitialized memory.